src/storage/v3/shard_rsm.cpp

```diff
@@ -51,6 +51,26 @@ using conversions::ToPropertyValue;
 namespace {
 namespace msgs = msgs;
 
+using AllEdgePropertyDataSructure = std::map<PropertyId, msgs::Value>;
+using SpecificEdgePropertyDataSructure = std::vector<msgs::Value>;
+
+using AllEdgeProperties = std::tuple<msgs::VertexId, msgs::Gid, AllEdgePropertyDataSructure>;
+using SpecificEdgeProperties = std::tuple<msgs::VertexId, msgs::Gid, SpecificEdgePropertyDataSructure>;
+
+using SpecificEdgePropertiesVector = std::vector<SpecificEdgeProperties>;
+using AllEdgePropertiesVector = std::vector<AllEdgeProperties>;
+
+using EdgeAccessors = std::vector<memgraph::storage::v3::EdgeAccessor>;
+
+using EdgeFiller = std::function<bool(const EdgeAccessor &edge, bool is_in_edge, msgs::ExpandOneResultRow &result_row)>;
+using EdgeUniqunessFunction = std::function<EdgeAccessors(EdgeAccessors &&, memgraph::msgs::EdgeDirection)>;
+
+struct VertexIdCmpr {
+  bool operator()(const storage::v3::VertexId *lhs, const storage::v3::VertexId *rhs) const {
+    return lhs->primary_label < rhs->primary_label && lhs->primary_key < rhs->primary_key;
+  }
```

Comment:
I think this is not a correct implementation. Let's say we have two vertices:
1. VertexA: label id is 1, primary key is [5]
2. VertexB: label id is 2, primary key is [4]

Then check the two cases:
1. VertexA < VertexB => `1 < 2 && 5 < 4` => false
2. VertexB < VertexA => `2 < 1 && 4 < 5` => false

I think the correct implementation would be `std::tie(lhs->primary_label, lhs->primary_key) < std::tie(rhs->primary_label, rhs->primary_key)`. See [`std::tie`](https://en.cppreference.com/w/cpp/utility/tuple/tie) and the [operators](https://en.cppreference.com/w/cpp/utility/tuple/operator_cmp) for `std::tuple`.

And the cherry on the cake would be to define this as `operator<` for `v3::VertexId` and use that function here as `*lhs < *rhs`.